libs/cloud/odc/aws/queue.py

```diff
@@ -180,3 +181,30 @@ def get_messages(
         raise Exception(f"Limit {limit} is not valid.")
 
     return itertools.islice(messages, limit)
+
+
+def publish_to_topic(topic_name: str, action: str, stac: dict):
+    """
+    Publish 'added' or 'archived' action to the provided sns topic
+    """
+    sns = boto3.client("sns")
+    sns_topic = sns.create_topic(Name=topic_name)
+
+    stac = json.dumps(stac)
+    message = {
+        "action": action,
+        "stac": stac,
+    }
+    sns.publish(
+        TopicArn=sns_topic.get("TopicArn"),
+        Message=json.dumps(message),
```

Comment:
The `Message` payload is just the STAC document itself. The `MessageAttributes` are used to describe it, so they can be filtered on when receiving lots of messages.

Our convention is as follows:

```
sns.publish(Message=stac_document, MessageAttributes={
    "action": { "DataType": "String", "StringValue": action },
    "product": {
        "DataType": "String", 
        "StringValue": stac_document["odc:product"] if "odc:product" in stac_doc else stac_document["collection"]
    },
    "maturity": {
        "DataType": "String",
        "StringValue": stac_doc["properties"]["dea:dataset_maturity"]
    }
})
```

Other message attributes I've come across are: `bbox.ll_lon`, `bbox.ll_lat`, `bbox.ur_lon`, `bbox.ur_lat`, `datetime`, `cloudcover`.

Reply:
Changed